crates/ruff/src/rules/flake8_todo/rules.rs

```diff
@@ -0,0 +1,384 @@
+    fn message(&self) -> String {
+        format!("Missing space after colon in TODO")
+    }
+}
+
+// Capture groups correspond to checking:
+//      1. Tag (used to match against `TODO` in capitalization and spelling, e.g. ToDo and FIXME)
+//      2. Author exists (in parentheses) with 0 or 1 spaces between it and the tag, but before colon
+//      3. Colon exists after author
+//      4. Space exists after colon
+//      5. Text exists after space
+//
+// We can check if any of these exist in one regex. Capture groups that don't pick anything up
+// evaluate to `None` in Rust, so the capture group index will always correspond to its respective rule
+// whether the token has been found or not.
+//
+// Example:
+// ```python
+// # TODO(evanrittenhouse): This is a completely valid TODO
+// ```
+// will yield [Some("TODO") Some("evanrittenhouse"), Some(":"), Some(" "), Some("This is a completely valid TODO")], whereas
+// ```python
+// # ToDo this is completely wrong
+// ```
+// will yield [Some("ToDo"), None, None, Some(" "), Some("this is completely wrong")]. Note the
+// `Nones` for the colon and space checks.
+//
+// Note: Regexes taken from https://github.com/orsinium-labs/flake8-todos/blob/master/flake8_todos/_rules.py#L12.
+static TODO_REGEX: Lazy<Regex> = Lazy::new(|| {
+    Regex::new(r"^# {0,1}(?P<tag>[tT][oO][dD][oO]|BUG|FIXME|XXX)( {0,1}\(.*\))?(:)?( )?(.+)?$")
+        .unwrap()
+});
+
+static ISSUE_LINK_REGEX_SET: Lazy<RegexSet> = Lazy::new(|| {
+    RegexSet::new([
+        r#"^#\s*(http|https)://.*"#, // issue link
+        r#"^#\s*\d+$"#,              // issue code - like "003"
+        r#"^#\s*[A-Z]{1,6}\-?\d+$"#, // issue code - like "TDO-003"
+    ])
+    .unwrap()
+});
+
+static NUM_CAPTURE_GROUPS: usize = 5usize;
+static TODO_LENGTH: usize = 4usize;
+
+pub fn check_todos(
+    tokens: &[LexResult],
+    autofix: flags::Autofix,
+    settings: &Settings,
+) -> Vec<Diagnostic> {
+    let mut diagnostics: Vec<Diagnostic> = vec![];
+    let mut prev_token_is_todo = false;
+    let mut prev_token_todo_start = 2; // Default to 2, the position of "T" in a properly formed TODO
+
+    for (start, token, end) in tokens.iter().flatten() {
+        let diagnostics_ref = &mut diagnostics;
+        let range = Range::new(*start, *end);
+
+        let token_opt = match token {
+            Tok::Comment(s) => Some(s),
+            _ => None,
+        };
+
+        // Check for errors due to a missing link: TDO003.
+        if prev_token_is_todo {
+            if token_opt.is_some() && ISSUE_LINK_REGEX_SET.is_match(token_opt.unwrap()) {
+                prev_token_is_todo = false;
+                continue;
+            }
+
+            diagnostics_ref.push(Diagnostic::new(
+                MissingLinkInTodo,
+                Range::new(
+                    Location::new(start.row() - 1, prev_token_todo_start),
+                    Location::new(end.row() - 1, prev_token_todo_start + TODO_LENGTH),
+                ),
+            ));
```

Comment:
Do we ned to add a check for this at the end of the file. E.g. 

```python
# TODO
```

No link at the end. 

An alternative would be to use a peekable iterator and perform a lookahead to see if the next comment is only separated by whitespace from this comment (or a single line break even and whitespace), and contains an issue link